### PR TITLE
Drop PHP 5.6 and 7.0

### PR DESCRIPTION
**Pull request by webimpress**
Dropped PHP 5.6 and 7.0 support.
PR based on #195. It should be merged before #196, because #196 contains changes which requires PHP 7.

Ping @Ocramius

### CODE REVIEW

File: test/AbstractFactory/ConfigAbstractFactoryTest.php [hunk 1]
```diff
@@ -184,8 +184,8 @@ public function testExceptsWhenConfigNotSet()
     {
         $abstractFactory = new ConfigAbstractFactory();
         $serviceManager = new ServiceManager();
-        self::expectException(ServiceNotCreatedException::class);
-        self::expectExceptionMessage('Cannot find a config array in the container');
+        $this->expectException(ServiceNotCreatedException::class);
```
Comment (Ocramius) [inline test/AbstractFactory/ConfigAbstractFactoryTest.php]: These should stay as `self`

Comment (webimpress) [inline test/AbstractFactory/ConfigAbstractFactoryTest.php]: Why? Methods `expectException*` are not static. Please see:
https://github.com/sebastianbergmann/phpunit/blob/master/src/Framework/TestCase.php#L569

Comment (Ocramius) [inline test/AbstractFactory/ConfigAbstractFactoryTest.php]: Oh, totally misread this, sorry :-\

**Comment by Ocramius** [comment]
LGTM, thanks @webimpress!

**Pull request merged**